src/animation.js

```diff
@@ -130,13 +132,39 @@
         return 'pending';
       if (this.paused)
         return 'paused';
-      if (this.finished)
+      if (this._isFinished)
         return 'finished';
       return 'running';
     },
+    get finished() {
+      if (!this._finishedPromise) {
```

Comment:
I'm OK with creating these on-demand. I don't think demand for them will be high.